frontend/rolecall/src/app/cast/cast-drag-and-drop.component.ts

```diff
@@ -343,13 +355,49 @@ export class CastDragAndDrop implements OnInit {
 
   decrementDancerCount(positionIndex: number) {
     this.castPositions[positionIndex].castRows.pop();
-    this.castPositions[positionIndex].castSize -= 1;
+    this.castPositions[positionIndex].dancerCount -= 1;
   }
 
   incrementDancerCount(positionIndex: number) {
     this.castPositions[positionIndex].castRows.push({
-      subCastDancers: new Array(CAST_COUNT),
+      subCastDancers: new Array(this.castCount),
     });
-    this.castPositions[positionIndex].castSize += 1;
+    this.castPositions[positionIndex].dancerCount += 1;
+  }
+
+  private changeCastCount(change: number) {
+    const oldCastCount = this.castCount;
+    this.castCount += change;
+    const oldCastPositions = this.castPositions;
+    this.castPositions = [];
+    for (let positionIndex = 0; positionIndex < oldCastPositions.length; positionIndex++) {
+      const oldPosition = oldCastPositions[positionIndex];
+      const newPosition: UICastPosition = {
+        pos: oldPosition.pos,
+        dancerCount: oldPosition.dancerCount,
+        castRows: [],
+      };
+      for(let dancerIndex = 0; dancerIndex < oldPosition.castRows.length; dancerIndex++) {
+        const oldCastRow = oldPosition.castRows[dancerIndex];
+        const transferCount = Math.min(this.castCount, oldCastCount)
+        const castRow: UICastRow = {
+          subCastDancers: new Array(this.castCount) as UICastDancer[],
```

Comment:
Let's try to avoid using `as` to force casting. It's dangerous since it overrides the type-check system, which means if the logic is refactored in the future and is wrong, it will not be caught by compiler and will result in runtime exception.

Try this:
```
subCastDancers: Array.from<UICastDancer>({length: this.castCount}),
```

Reply:
Done.